core/modules/search/search.admin.inc

```diff
@@ -137,6 +137,15 @@ function search_admin_settings($form, &$form_state) {
     '#description' => t('Only one type of item will be searched by default. This selection will determine what appears at <a href="@search">@search</a>.', array('@search' => url('search', array('absolute' => TRUE)))),
   );
 
+  /* Add checkbox for partial word search */
+  $search_partial_words = $config->get('search_partial_words_enable');
```

Comment:
We can shorten the setting to just `search_partial_words`.

It will need to be added to `search.settings.json` and also set the default to `0` in an update hook in `search.install`.